core/primitives/src/shard_layout.rs

```diff
@@ -16,6 +23,8 @@ pub enum ShardLayout {
 pub struct ShardLayoutV0 {
```

Comment:
do these have to be public? Is there some place where we use `ShardLayoutV0` directly instead of `ShardLayout`?

Reply:
yeah, I think they do, because they can be accessed from ShardLayout directly

Reply:
error[E0446]: private type `ShardLayoutV1` in public interface